src/resolve/resolved-conf.c

```diff
@@ -128,6 +130,100 @@ int manager_parse_search_domains_and_warn(Manager *m, const char *string) {
         }
 }
 
+int manager_init_dnssec_algorithms(Manager *m) {
+        int r;
+
+        assert(m);
+
+        r = set_ensure_allocated(&m->dnssec_algorithms, NULL);
+        if (r < 0)
+                return r;
+
+#if HAVE_OPENSSL_OR_GCRYPT
+        for (int i = 0; i < _DNSSEC_ALGORITHM_MAX_DEFINED; i++) {
+                hash_md_t algorithm;
+
+                algorithm = algorithm_to_implementation_id(i);
+#if PREFER_OPENSSL
+                if (!algorithm)
+                        continue;
+#else
+                if (algorithm == -EOPNOTSUPP)
```

Comment:
i think any kind of error should result in the entry being skipped, no? i.e. generalize check to `< 0`?